bin/mqtt5_canary/main.cpp

```diff
@@ -450,7 +450,7 @@ static int s_AwsMqtt5CanaryOperationUnsubscribeBad(struct AwsMqtt5CanaryTestClie
             unsubscription, [testClient](int, std::shared_ptr<Mqtt5::UnSubAckPacket> packet) {
                 if (packet == nullptr)
                     return;
-                if (packet->getReasonCodes()[0] == AWS_MQTT5_UARC_SUCCESS)
+                if (packet->getReasonCodes()[0] == UnSubAckReasonCode::AWS_MQTT5_UARC_SUCCESS)
```

Comment:
This isn't what I had in mind for the enums.  Did the original proposal not work?

Reply:
As I'm creating the enum using the same value name in aws-c-mqtt (`UnSubAckReasonCode:: AWS_MQTT5_UARC_SUCCESS` v.s. `aws_mqtt5_unsuback_reason_code::AWS_MQTT5_UARC_SUCCESS`), the compiler would report an ambiguity error. I had to add the namespace to avoid the error.

One way to fix it is using a different enum value name in CPP. For example: `UnSubAckReasonCode ::AWS_MQTT5_UARC_SUCCESS` -> `UnSubAckReasonCode::SUCCESS`. In this case, we could still directly pass `AWS_MQTT5_UARC_SUCCESS ` as the enum would be converted implicitly. However, it would break the user who was using `UnSubAckReasonCode::AWS_MQTT5_UARC_SUCCESS`. 

I'm not sure if there is a better way to fix it.

Reply:
That was what I intended:

enum class UnsubackReasonCode {
   Success = AWS_MQTT5_UARC_SUCCESS,
   ... etc...
};

Reply:
1. It seems that enum class would not be able to directly convert between int and enum value. Using enum class would definitely breaks the customer. 

2. If we removed AWS_MQTT5_UARC_SUCCESS and rename it to `Success`, then it would break the customer if they already use the enum with the namespace like following pattern:
```cpp
UnSubAckReasonCode reasonCode = UnSubAckReasonCode::AWS_MQTT5_UARC_SUCCESS
```

Reply:
You could double the entries with the palatable names, but I'm not sure the extra effort is worth it though.

Forcing people to use the name mangling prefix is crummy  though.